File: Geeks for Geeks/linkedlist.py
class linklist:
    def __init__(self):
        self.head = None
        self.tail = None

    def append(self, newNode):
        if self.head == None:
            self.head = newNode
            self.tail = newNode
            return
        self.tail.next = newNode
        self.tail = self.tail.next

    def display(self):
        nextNode = self.head
        while nextNode:
            print(nextNode.data, end=" ")
            nextNode = nextNode.next


class node:
    def __init__(self, data):
        self.data = data
        self.next = None


def main():
    ll = linklist()

    n = int(input())
    arr = list(map(int, input().split()))

    for num in arr:
        newNode = node(num)
        ll.append(newNode)
    print(ll.display())


if __name__ == "__main__":
    main()

File: CTCI/linkedList.js
class Node{
    constructor(data){
        this.data = data;
        this.next = null;
    }
}

class LinkedList{

    constructor(){
        this.head = null;
    }

    insertEnd(data){

        let newNode = new Node(data);

        if(this.head === null){
            this.head = newNode;
            return
        }

        let curr = this.head;

        while (curr.next)
            curr = curr.next;

        curr.next = newNode;
    }

    insertStart(data){
        let newNode = new Node(data);
        if (this.head === null){
            this.head = newNode;
            return
        }
        newNode.next = this.head;
        this.head = newNode;
    }

    deleteIndex(ind){
        let curr = this.head;
        if (ind === 1){
            this.head = this.head.next;
            return
        }
        let iter = 0;
        while (curr){
            if ( iter + 1 === ind ){
                curr.next = curr.next.next;
                return true
            }
            curr = curr.next;
            iter++;
        }
        return false
    }

    deleteNode(data){
        if (this.head.data === data){
            this.head = this.head.next;
            return
        }
        let curr = this.head;
        while (curr.next){
            if(curr.next.data === data){
                curr.next = curr.next.next;
                return true
            }
            curr = curr.next;
        }
        return false
    }

    display(){
        let curr = this.head;
        let ans = "";
        while (curr){
            ans += " " + curr.data;
            curr = curr.next
        }
        return ans;
    }
}

export { LinkedList, Node }
